src/amuse/units/core.py

```diff
@@ -1046,8 +1042,7 @@ def reference_string(self):
     @late
     def base(self):
         return tuple(
-            filter(lambda x: x[0] != 0,
-            map(lambda x : (x[0] * self.power, x[1]), self.local_unit.base)))
+            [x for x in [(x[0] * self.power, x[1]) for x in self.local_unit.base] if x[0] != 0])
```

Comment:
also needs parentheses to be human readable

Reply:
fixed